frontend/views/components/ButtonSubmit.vue

```diff
@@ -0,0 +1,50 @@
+<template lang='pug'>
+  button.is-loader(
+    v-bind='$attrs'
+    v-on='bindListeners'
+    type='submit'
+    :data-loading='ephemeral.isSubmitting'
+    :disabled='disabled || ephemeral.isSubmitting'
+  )
+    slot
+</template>
+
+<script>
+export default {
+  name: 'ButtonSubmit',
+  props: {
+    disabled: Boolean
+  },
+  data: () => ({
+    ephemeral: {
+      isSubmitting: false
+    }
+  }),
+  methods: {
+    async submit (e) {
+      if (this.ephemeral.isSubmitting) {
+        return
+      }
+
+      this.ephemeral.isSubmitting = true
+
+      // this.$listeners can await for async events.
+      // An advantage over $emit()
+      // More at: https://stackoverflow.com/questions/60554270/vuejs-difference-between-emit-and-listeners
+      await this.$listeners.click(e)
+
+      this.ephemeral.isSubmitting = false
```

Comment:
So here we need to be careful, because if an exception is thrown on `.click(e)`, then `isSubmitting` will not be reset back to `false`, so we need to wrap the call to `.click` in a try/catch.

```js
try {
  await this.$listeners.click(e)
} catch (err) {
  console.error('ButtonSubmit exception:', err)
}

this.ephemeral.isSubmitting = false
```